cdap-ui/app/cdap/components/PipelineDetails/PipelineRuntimeArgsDropdownBtn/RuntimeArgsKeyValuePairWrapper/RuntimeArgsPairsMaterial.tsx

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import * as React from 'react';
+import AbstractMultiRowWidget, {
+  IMultiRowProps,
+} from 'components/AbstractWidget/AbstractMultiRowWidget';
+import RuntimeArgsRow from 'components/PipelineDetails/PipelineRuntimeArgsDropdownBtn/RuntimeArgsKeyValuePairWrapper/RuntimeArgsRow';
+import ThemeWrapper from 'components/ThemeWrapper';
+import { objectQuery } from 'services/helpers';
+
+interface IRuntimeArgsPairsWidgetProps {
+  'key-placeholder'?: string;
+  'value-placeholder'?: string;
+  'kv-delimiter'?: string;
+  delimiter?: string;
+}
+
+interface IRuntimeArgsPairsProps extends IMultiRowProps<IRuntimeArgsPairsWidgetProps> {
+  isEncoded?: boolean; // for compatiblity with keyvalue-encoded type
+}
+
+class RuntimeArgsPairsView extends AbstractMultiRowWidget<IRuntimeArgsPairsProps> {
+  public componentWillReceiveProps(nextProps) {
+    const currentValue = this.constructValues();
+    if (currentValue.length === nextProps.value.length) {
+      return;
+    }
+    this.init(nextProps);
+  }
+  public constructValues = () => {
+    const values = this.state.rows
+      .filter((id) => this.values[id] && this.values[id].value)
+      .map((id) => this.values[id].value);
+    return values;
+  };
+  public deconstructValues = (props) => {
+    if (!props.value || props.value.length === 0) {
+      return [];
+    }
+    const kvDelimiter = objectQuery(props, 'widgetProps', 'kv-delimiter') || ':::';
```

Comment:
Where is this `:::` coming from? I thought we decided to not use a string delimiter as we want to pass in the runtime arguments as is (object).